src/blockchain/rpc.rs

```diff
@@ -441,6 +441,59 @@ fn list_wallet_dir(client: &Client) -> Result<Vec<String>, Error> {
     Ok(result.wallets.into_iter().map(|n| n.name).collect())
 }
 
+/// Factory of [`RpcBlockchain`] instances, implements [`BlockchainFactory`]
+///
+/// Internally caches the node url and authentication params and allows getting many different [`RpcBlockchain`]
+/// objects for different wallet names and with different rescan heights.
+///
+/// ## Example
+///
+/// ```no_run
+/// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// let factory = RpcBlockchainFactory {
+///     url: "http://127.0.0.1:18332",
+///     auth: Auth::Cookie {
+///         file: "/home/user/.bitcoin/.cookie".to_string(),
+///     },
+///     network: Network::Testnet,
+///     wallet_name_prefix: "prefix-".to_string(),
+///     default_skip_blocks: 100_000,
+/// };
+/// let main_wallet_blockchain = factory.build("main_wallet", Some(200_000))?;
+/// # Ok(())
+/// ```
+#[derive(Debug, Clone)]
+pub struct RpcBlockchainFactory {
+    /// The bitcoin node url
+    pub url: String,
+    /// The bitcoin node authentication mechanism
+    pub auth: Auth,
+    /// The network we are using (it will be checked the bitcoin node network matches this)
+    pub network: Network,
+    /// The prefix used to build the full wallet name for blockchains
```

Comment:
I don't understand the purpose of `wallet_name_prefix` if we use unique wallet names. Or i guess this is just in case your core node happens to have a wallet by the same name?

Reply:
It's true that it's not entirely required, but I think it could be useful to group all the sub-wallets that belong to the same "main wallet" without having to keep track of the prefix explicitly.

I'm gonna make it an `Option`, so if you are not interested in it you can just use `None` there